embeddings/task/lightning_task/text_classification.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, Optional, Tuple, Type
+
+import numpy as np
+import torch
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from torch.utils.data import DataLoader
+from torchmetrics import F1, Accuracy, MetricCollection, Precision, Recall
+from transformers import AutoModel, AutoModelForSequenceClassification
+
+from embeddings.data.datamodule import HuggingFaceDataset
+from embeddings.task.lightning_task.lightning_task import HuggingFaceLightningTask
+
+
+class TextClassificationTask(HuggingFaceLightningTask):
+    def __init__(
+        self,
+        num_labels: int,
+        model_name_or_path: str,
+        downstream_model_type: Type["AutoModel"] = AutoModelForSequenceClassification,
+        unfreeze_transformer_from_layer: Optional[int] = None,
+        metrics: Optional[MetricCollection] = None,
+        config_kwargs: Optional[Dict[str, Any]] = None,
+        task_model_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(
+            num_labels=num_labels,
+            model_name_or_path=model_name_or_path,
+            downstream_model_type=downstream_model_type,
+            unfreeze_transformer_from_layer=unfreeze_transformer_from_layer,
+            metrics=metrics,
+            config_kwargs=config_kwargs,
+            task_model_kwargs=task_model_kwargs,
+        )
+
+    def get_default_metrics(self) -> MetricCollection:
+        if self.hparams.num_labels > 2:
+            metrics = MetricCollection(
+                [
+                    Accuracy(num_classes=self.hparams.num_labels),
+                    Precision(num_classes=self.hparams.num_labels, average="macro"),
```

Comment:
Do we prefer `macro` or `weighted`? I used mostly `weighted` average due to possible imbalances in data. wdyt? @ktagowski @albert097 @riomus @djaniak

Reply:
I think that is task-dependent it is a choice whether we need to weight class importance. It is mainly for purpose of optimization step of the model (schedulers, early stopping etc.) After the training is finished instead of these metrics defined here we use our Evaluators classes.

Reply:
I usually prefer macro-average since it's better to assume that there may be imbalance in the data and then the metric will tell us that the predictor doesn't work on the particular class. Although if we were to use these metrics in a sequence tagging task I'd probably prefer the weighted average since we will probably have more not fully covered classes in that case.